app/src/main/java/fr/neamar/kiss/dataprovider/AppProvider.java

```diff
@@ -19,30 +20,68 @@ public void reload() {
         ArrayList<Pojo> records = new ArrayList<>();
 
         int relevance;
-        int matchPositionStart;
-        int matchPositionEnd;
         String appNameNormalized;
 
-        final String queryWithSpace = " " + query;
         for (Pojo pojo : pojos) {
             relevance = 0;
             appNameNormalized = pojo.nameNormalized;
 
-            matchPositionEnd = 0;
-            if (appNameNormalized.startsWith(query)) {
-                relevance = 100;
-                matchPositionStart = 0;
-                matchPositionEnd = query.length();
-            } else if ((matchPositionStart = appNameNormalized.indexOf(queryWithSpace)) > -1) {
-                relevance = 50;
-                matchPositionEnd = matchPositionStart + queryWithSpace.length();
-            } else if ((matchPositionStart = appNameNormalized.indexOf(query)) > -1) {
-                relevance = 1;
-                matchPositionEnd = matchPositionStart + query.length();
+            ArrayList<Pair<Integer, Integer>> matchPositions = new ArrayList<>();
```

Comment:
So putting it as a member of the class? Because if I understand correctly the function getResults is called on each keystroke, so that's the only way?